### PR TITLE
Build out REST Endpoints for Resources 

### PR DESCRIPTION
**Pull request by Melvin1Atieno**
#### What does this PR do?
Have CRUD on resources endpoint for MnR tracker REST API

#### Description of Task to be completed?
Have the following endpoints
1. GET  /api/v1/Users/requests
2. GET   /api/v1/users/requests/<int:id>
3. PUT  /api/v1/users/requests/<int:id>
4. POST  /api/v1/users/requests/
5 .POST  /api/v1/users/registration
6 .POST /api/v1/users/
#### How should this be manually tested?
1. clone repo.
2. checkout Develop.
3. activate virtual env
4. pip install requirements .txt
5. flask run
. using postman test every endpoint with this content header
key: Content-Type value : application/json
#### Screenshots
Sample from postman
postman sample

### CODE REVIEW

**Comment by Slinjez** [comment]
PR nicely done

File: app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    request_category = request.json.get("request_category")
+                    
+                    updated_request = {
+                        "request_id": request_id,
+                        "request_title": request_title,
+                        "request_description": request_description,
+                        "request_category": request_category
+                    }
+    
+                    request_catalog.remove(request_details)
```
Comment (Tittoh) [inline app/__init__.py]: You could update the dictionary without removing the contents.

Comment (Tittoh) [inline app/__init__.py:161]: You could update the dictionary without removing the contents.

File: run.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from app import app
+
+def create_app(config_filename):
+    app = Flask(__name__)
+
+    app_config.from_object(config_filename)
+
+    api
+
+
```
Comment (Tittoh) [inline run.py:14]: Consider PEP 8 style guide for blank lines to remove unnecessary ones.

File: tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        #create a request
+        create = self.client.post("/api/v1/users/requests",
+        data=json.dumps(self.data),
+        content_type=("application/json")
+        )
+        #assert request has been created 
+        self.assertEqual(create.status_code,201)
+#update the request
+        change = self.client.put("/api/v1/users/requests/1",
+        data=json.dumps(dict(request_title ="kitchen",request_description="whitescreen",request_category="Repair")), content_type=("application/json"))
```
Comment (Tittoh) [inline tests/test_api.py:92]: Break these long lines

File: tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        response_msg = json.loads(response.data.decode())
+        self.assertEqual(response_msg["message"],"successfully logged in")
+        self.assertEqual(response.status_code,200)
+
+
+    
+        
+        
+        if __name__ == "__main__":
+            unittest.main()
```
Comment (Tittoh) [inline tests/test_api.py:154]: the code is elegant

Comment (Melvin1Atieno) [reply tests/test_api.py:154]: @Tittoh Thank you. Let me get to it.

**Pull request merged**